inffast.c

```diff
@@ -176,31 +189,19 @@ void Z_INTERNAL zng_inflate_fast(PREFIX3(stream) *strm, unsigned long start) {
         } else if (op & 16) {                     /* length base */
             len = here->val;
             op &= 15;                           /* number of extra bits */
-            if (bits < op) {
-                hold |= load_64_bits(in, bits);
-                in += 6;
-                bits += 48;
-            }
             len += BITS(op);
             DROPBITS(op);
             Tracevv((stderr, "inflate:         length %u\n", len));
-            if (bits < 15) {
-                hold |= load_64_bits(in, bits);
-                in += 6;
-                bits += 48;
-            }
             here = dcode + (hold & dmask);
+            if (bits < 15 + 13) {
```

Comment:
I would avoid addition of two numeric literals as much as possible in conditional statements and blocks. 

Even though compiler automatically does the addition before emitting the code, it makes the code less clean. Avoiding them also makes it clear for a compiler that the developer knows how many bits the calculation needs without truncating the result.

Reply:
I think it's ok. There is code all over the world that does stuff like this and probably even in our source code we can find instances where macros are defined like that.

Reply:
Should these magic numbers be defined instead?

Reply:
Ideally yes, but magic numbers were used for this before, and I'd prefer to keep these PRs small so other zlibs can pick them up, rather than mixing in code style changes. (If someone else wants to refactor in a separate PR, they're constants from the DEFLATE specification: the maximum length of a distance huffman code (15, defined as `MAXBITS` in `inftrees.c`), plus the maximum number of extra distance bits (13).)

Reply:
It's not hard to add those "meanings" as a comment... Doing in that way makes reviewing code now and later a little bit easier...

zlib-ng doesn't follow coding style of Mark Adler's original zlib repository as style of zlib has to be compatible with very old compilers and screen resolutions. As I've mentioned before, zlib-ng only supports compilers that are up to about 7 years old (Visual C++ 2015 is oldest supported for Windows). We have some workarounds for missing intrinsic functions in older VC++, gcc and clang versions, but we try to keep them as minimum as possible. Oldest gcc and clang versions I use for daily work are from 6.x series, but there is supported platforms that still need gcc 4.8 or gcc 4.9.

Reply:
> I think it's ok. There is code all over the world that does stuff like this and probably even in our source code we can find instances where macros are defined like that.

Defining macros is quite different issue as macros go through the preprocessor first. Preprocessor has no concept of integer widths or signedness. Even macros have limitations as I've mentioned before, including lack of deduplication and as such evaluating some expressions more than once, or mandatory requirement of using parentheses around variable names even when there is clearly defined order of operator precedence due to expanding variable contents as string instead of evaluating and replacing variable name with result.

Reply:
I do prefer that we add comments describing where they came from when refactoring code using "magic numbers", or even add them as named defines, especially when we use them multiple places. Researching what some of these numbers mean is sometimes nontrivial and the person working on the refactoring has often already figured out their meaning, so "sharing is caring" is the correct mindset here I guess 😃.

This is just so it is easier for others to read and/or review the code, and to avoid mistakes where someone assumes a number has a certain meaning just because it is the same number as was used elsewhere (we have had confusions of this kind a few times before, sometimes resulting in corner-case bugs).